pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java

```diff
@@ -252,6 +258,25 @@ public boolean verify(String hostname, SSLSession session) {
     }
     // Extract the common name
     X509Certificate serverCert = peerCerts[0];
+
+    try {
+      // Check for Subject Alternative Names (see RFC 6125)
+      Collection subjectAltNames = serverCert.getSubjectAlternativeNames();
```

Comment:
If there is no SAN list  in the cert then this would return null which would lead to an NPE. Need to add a `... !=null` check and if null then skip SAN validation.